src/Functions/FunctionsHashing.h

```diff
@@ -1126,15 +1126,18 @@ class FunctionAnyHash : public IFunction
             size_t size = vec_from.size();
             for (size_t i = 0; i < size; ++i)
             {
-                ToType hash;
-                if constexpr (std::endian::native == std::endian::little)
-                    hash = apply(key, reinterpret_cast<const char *>(&vec_from[i]), sizeof(vec_from[i]));
-                else
+                // Take a copy to maintain column representation as is.
+                auto value = vec_from[i];
+                transformEndianness<std::endian::little>(value);
+
+                if constexpr (std::is_same_v<FromType, UUID>)
                 {
-                    char tmp_buffer[sizeof(vec_from[i])];
-                    reverseMemcpy(tmp_buffer, &vec_from[i], sizeof(vec_from[i]));
-                    hash = apply(key, reinterpret_cast<const char *>(tmp_buffer), sizeof(vec_from[i]));
+                    auto & uuid = value.toUnderType();
+                    std::swap(uuid.items[0], uuid.items[1]);
```

Comment:
This exact operation reoccurs a couple of times throughout `FunctionsHashing.h` and `reinterpretAs.cpp`. Perhaps you could make it a common function?

Reply:
Added a function to the UUID helpers for doing this swap.